src/fdc3/dotnet/DesktopAgent/src/DesktopAgent/Fdc3DesktopAgent.cs

```diff
@@ -124,26 +126,40 @@ public async ValueTask<FindIntentResponse> FindIntent(FindIntentRequest? request
             return FindIntentResponse.Failure(ResolveError.IntentDeliveryFailed);
         }
 
-        //This function returns null, if the app could not be accepted based on the intent (required), context (optional in request), resultType (optional in request)
-        //else for consistency it will return a single element array containing the intentMetadata which is allowed by the request.
-        Func<Fdc3App, IEnumerable<IntentMetadata>?> selector = (fdc3App) =>
+        var appIntents = new Dictionary<string, AppIntent>();
+
+        void FindIntentQuery(Fdc3App fdc3App, string? instanceId)
```

Comment:
Why did you change the return type to `void`? To me a *Query returns something. Similarly the `Get**` methods below should return something, or they should have a different name.

Reply:
But why? If I look at a method which is called `GetSomething()` I'd expect it to return `Something`, and not cause any other side effects.

Reply:
Actually I thought that it is more cleaner, but I can revert that back